TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/vision/ComputerVisionProcessor.java

```diff
@@ -50,57 +49,68 @@ public Object processFrame(Mat inputFrameRGB, long captureTimeNanos) {
         if (inputFrameRGB.type() == 24) {
             Imgproc.cvtColor(inputFrameRGB, inputFrameRGB, Imgproc.COLOR_RGBA2RGB);
         }
-        leftMat.update(inputFrameRGB);
         middleMat.update(inputFrameRGB);
+        rightMat.update(inputFrameRGB);
         framesProcessed++;
-        setPropLocation();
-        return inputFrameRGB; // Don't think this does anything
+        cameraTelemetry();
+        detectPropLocation();
+
+
```

Comment:
Another option that may work is to manually call garbage collection here. That will take processing time, but should prevent the memory from ramping up.